src/test/java/io/confluent/idesidecar/restapi/resources/ConnectionsResourceTest.java

```diff
@@ -469,6 +477,102 @@ void listConnections_returnsAllConnections() {
     assertConnectionList(expectedList, actualList);
   }
 
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("partialPayloadParams")
+  void updateConnectionViaPatchWithPartialPayloadsShouldSucceed(
+      String testName,
+      String requestId,
+      String requestName,
+      String requestType) throws Exception {
+
+    // Create authenticated connection
+    var connectionId = "c1";
+    var connectionName = "Connection 1";
+    var orgName = "Development Org";
+    var accessTokens = ccloudTestUtil.createAuthedCCloudConnection(
+        connectionId,
+        connectionName,
+        orgName,
+        null
+    );
+    var connectionSpec = connectionStateManager.getConnectionSpec("c1");
+
+    var testContext = new VertxTestContext();
+    refreshConnectionStatusAndThen(connectionId, testContext, () -> {
+      assertAuthStatus(connectionId, "VALID_TOKEN");
+
+      // Log current state if testing type change
+      if (testName.contains("Change Type")) {
```

Comment:
This way we don't have to rely on the `testName` for determining if the test case updates the connection type:

```suggestion
    var connectionSpec = connectionStateManager.getConnectionSpec("c1");
    var testUpdatesConnectionType = !connectionSpec.type().equals(
        ConnectionType.valueOf(requestType));

    var testContext = new VertxTestContext();
    refreshConnectionStatusAndThen(connectionId, testContext, () -> {
      assertAuthStatus(connectionId, "VALID_TOKEN");

      // Log current state if testing type change
      if (testUpdatesConnectionType) {
```